src/scripts/themed_posts.js

```diff
@@ -24,41 +29,67 @@ const processPosts = async function (postElements) {
   filterPostElements(postElements, { includeFiltered: true }).forEach(async postElement => {
     if (postElement.matches(blogViewSelector) && !enableOnPeepr) return;
 
-    const { blog: { name, theme } } = await timelineObject(postElement);
-
-    if (blacklist.includes(name)) return;
-
-    if (!blogs.has(name)) {
-      blogs.add(name);
-
-      const {
-        backgroundColor,
-        titleColor,
-        linkColor
-      } = theme;
-
-      const backgroundColorRGB = hexToRGB(backgroundColor);
-      const titleColorRGB = hexToRGB(titleColor);
-      const linkColorRGB = hexToRGB(linkColor);
-
-      styleElement.textContent += `
-        [data-xkit-themed="${name}"] {
-          --white: ${backgroundColorRGB};
-          --black: ${titleColorRGB};
-          --accent: ${linkColorRGB};
-          --color-primary-link: rgb(var(--accent));
-        }
-      `;
+    const { blog, trail = [] } = await timelineObject(postElement);
+
+    const blogData = [
+      blog,
+      ...reblogTrailTheming ? trail.map(item => item.blog) : []
+    ];
+
+    blogData.forEach(({ name, theme }) => {
+      if (blacklist.includes(name)) return;
+
+      if (!blogs.has(name)) {
+        blogs.add(name);
+
+        const {
+          backgroundColor,
+          titleColor,
+          linkColor
+        } = theme;
+
+        const backgroundColorRGB = hexToRGB(backgroundColor);
+        const titleColorRGB = hexToRGB(titleColor);
+        const linkColorRGB = hexToRGB(linkColor);
+
+        styleElement.textContent += `
+          [data-xkit-themed="${name}"] {
+            --white: ${backgroundColorRGB};
+            --black: ${titleColorRGB};
+            --accent: ${linkColorRGB};
+            --color-primary-link: rgb(var(--accent));
+          }
+        `;
+      }
+    });
+
+    postElement.dataset.xkitThemed = blog.name ?? '';
+
+    if (reblogTrailTheming) {
+      [...postElement.querySelectorAll(reblogSelector)].forEach((reblog, i) => {
+        reblog.dataset.xkitThemed = trail[i]?.blog?.name ?? '';
+      });
     }
-
-    postElement.dataset.xkitThemed = name;
   });
 };
 
 export const main = async function () {
-  ({ enableOnPeepr, blacklistedUsernames } = await getPreferences('themed_posts'));
+  ({ reblogTrailTheming, enableOnPeepr, blacklistedUsernames } = await getPreferences('themed_posts'));
   blacklist = blacklistedUsernames.split(',').map(username => username.trim());
 
+  reblogSelector = await keyToCss('reblog');
+
+  styleElement.textContent += `
+    article ${reblogSelector} {
+      overflow-y: hidden;
+      margin-top: 0;
+    }
```

Comment:
```suggestion
  styleElement.textContent += `
    article ${reblogSelector} {
      display: flow-root;
      margin-top: 0;
    }
```

Does this serve the same purpose as setting the overflow? I saw it in the infobox in https://www.joshwcomeau.com/css/rules-of-margin-collapse/#margins-can-collapse-in-the-same-direction - never heard of it before.

Reply:
Ah, once again my instinct about more simple things existing turns out to be right. I wouldn't have thought it was a `display` value...

Tested and works.

Reply:
Me neither! I feel like it should have its own MDN. The caniuse looks fine, though.